cmd/dp-dataset-exporter/main.go

```diff
@@ -106,28 +129,64 @@ func main() {
 		vaultClient,
 		datasetAPICli,
 	)
+	services[serviceHealthTicker] = true
 
-	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+	// Gracefully shutdown the application closing any open resources
+	gracefulShutdown := func() {
+		log.Info("gracefully shutting down", log.Data{"graceful_shutdown_timeout": cfg.GracefulShutdownTimeout})
+		ctx, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
+
+		if services[serviceConsumer] {
+			err = eventConsumer.Close(ctx)
+			logIfError(err)
```

Comment:
_über_ nit-picky but could we not just make the close call directly from within the log func?
```
logIfError(eventConsumer.Close(ctx))
```

Saves unnecessary lines and var assignments. (sorry very nit-picky)